src/core/lib/slice/slice_internal.h

```diff
@@ -327,4 +327,6 @@ inline grpc_slice grpc_slice_from_static_string_internal(const char* s) {
   return grpc_slice_from_static_buffer_internal(s, strlen(s));
 }
 
+grpc_slice grpc_slice_split_head_noref(grpc_slice* source, size_t split);
```

Comment:
I'll make my usual comment here: Can we avoid adding more variants to these APIs that make it harder for callers to understand how to use it?

I really think we need to take a step back and re-think our approach here.  We need to restructure these APIs such that they are simple to use and still have the information they need to automatically pick the most efficient implementation in each case.

Reply:
This is analogous to this existing method:

/** The same as grpc_slice_split_tail, but with an option to skip altering
 * refcounts (grpc_slice_split_tail_maybe_ref(..., true) is equivalent to
 * grpc_slice_split_tail(...)) */
GPRAPI grpc_slice grpc_slice_split_tail_maybe_ref(grpc_slice* s, size_t split,
                                                  grpc_slice_ref_whom ref_whom);

There's no way to really encode type information here to select a method overload, as far as I know.